### PR TITLE
Fix parsing backslash escaped glyph names in features

### PR DESCRIPTION
**Pull request by anthrotype**
Fixes https://github.com/googlei18n/fontmake/issues/445

### CODE REVIEW

File: tests/writer_test.py [hunk 1]
```diff
@@ -1025,7 +1025,7 @@ def test_write_node(self):
             '"130 431 LINE {\\"\\012key\\\\"\';\\012\\012\\012\\" '
             '= \\"\\\\"\'value\\012several lines\\012;\\012\\"'
             ';\\n\\";\\" = \\";\\012\\";\\n'
-            'escapeception = \\"\\\\\\\\"\\\\\'\\\\n\\\\\\\\n\\";}"'
+            'escapeception = \\"\\\\\\\\"\\\\\'\\\\\\n\\\\\\\\\\n\\";}"'
```
Comment (belluzj) [inline tests/writer_test.py]: Can you check that your new value is indeed the output of the latest Glyphs version? I remember getting surprising results... So can you run the code above to create a point and assign userData in Glyphs and save the file, to check which output you get.

Comment (anthrotype) [reply tests/writer_test.py]: I don’t have a Mac at hand, I was hoping one of you guys would try and check this.
I spent the whole day scratching my head around this and this is the only way that i could make it round trip with arbitrary level of nesting and escaping.

Comment (madig) [reply tests/writer_test.py]: I get this string written to disk from glyphs:
```
nodes = (
"404 387 LINE {\"\012key\\"';\012\012\012\" = \"\\"'value\012several lines\012;\012\";\n\";\" = \";\012\";\nescapeception = \"\\\\"\\'\\n\\\\n\";}",
"209 390 LINE",
"299 619 LINE"
);
```
Now to test for equality...

Comment (madig) [reply tests/writer_test.py]: Glyph writes the string that was before your change :(

Comment (madig) [reply tests/writer_test.py]: Opening the file above again in Glyphs makes a slash disappear:
```
{
    "\nkey\"';\n\n\n" = "\"'value\nseveral lines\n;\n";
    ";" = ";\n";
    escapeception = "\\\"\\'\n\\\n";
}
```
@schriftgestalt :)

**Comment by schriftgestalt** [comment]
I checked this. When I run your python script, the .glyphs file contains some `\n` to the end of the node string. That is "fixed", when saving and reopening the file. But all slashes stay the same.

**Pull request merged**